lib/swim.js

```diff
@@ -250,8 +252,169 @@ PingSender.prototype.doCallback = function doCallback(isOk, bodyObj) {
     }
 };
 
+function Gossip(ringpop) {
+    this.ringpop = ringpop;
+    this.isStopped = null;
+    this.timer = null;
+}
+
+Gossip.prototype.start = function start() {
+    if (this.isStopped === true) {
+        this.ringpop.logger.debug('gossip was not started because it was previously stopped. use `restart()`', {
+            local: this.ringpop.membership.getLocalMemberAddress()
+        });
+        return;
+    }
+
+    var self = this;
+    var startTime = new Date();
+
+    var protocolDelay = this.ringpop.computeProtocolDelay();
+
+    this.ringpop.stat('timing', 'protocol.delay', protocolDelay);
+
+    this.timer = setTimeout(function() {
+        self.ringpop.pingMemberNow(function() {
+            self.ringpop.stat('timing', 'protocol.frequency', startTime);
+            self.start();
+        });
+    }, protocolDelay);
+
+    this.isStopped = false;
+};
+
+Gossip.prototype.stop = function stop() {
+    if (this.isStopped !== false) {
+        this.ringpop.logger.warn('cannot stop gossip protocol because it was never started', {
+            local: this.ringpop.membership.getLocalMemberAddress()
+        });
+        return;
+    }
+
+    clearTimeout(this.timer);
+    this.timer = null;
+
+    this.isStopped = true;
+
+    this.ringpop.logger.debug('stopped gossip protocol', {
+        local: this.ringpop.membership.getLocalMemberAddress()
+    });
+};
+
+Gossip.prototype.restart = function restart() {
+    if (this.isStopped !== true) {
+        this.ringpop.logger.warn('cannot restart gossip protocol because it was never stopped', {
```

Comment:
A restart method makes sense in both `STOP` and `START` state.

It should be `if (started) { stop() }; start();`

Otherwise this mehtod might as well be called `.start()`